database/pgsql/layer.go

```diff
@@ -238,28 +250,51 @@ func (pgSQL *pgSQL) InsertLayer(layer database.Layer) error {
 
 	// Get parent ID.
 	var parentID zero.Int
+	mapNamespaceIDs := make(map[string]int)
 	if layer.Parent != nil {
 		if layer.Parent.ID == 0 {
 			log.Warning("Parent is expected to be retrieved from database when inserting a layer.")
 			return cerrors.NewBadRequestError("Parent is expected to be retrieved from database when inserting a layer.")
 		}
 
 		parentID = zero.IntFrom(int64(layer.Parent.ID))
-	}
 
-	// Find or insert namespace if provided.
-	var namespaceID zero.Int
-	if layer.Namespace != nil {
-		n, err := pgSQL.insertNamespace(*layer.Namespace)
+		// Find its parent's namespaces
+		rows, err := pgSQL.Query(searchLayerNamespace, parentID)
```

Comment:
The layer.Parent.Namespaces are loaded in work.go.
But in layer_test, it is not. In FindLayer (around L55), the parent's ID and name are loaded, but parent's namespace is not.

As we already make the assumption that the provided Parent layer has been retrieved.
So either we modify the related test cases or load parent's namespace in FindLayer.
My second commits did the later choice, but it will be time consuming.